File: backend/src/lib/PriceCalculationRequest.ts
import Attribute from "./product/Attribute";

export default class PriceCalculationRequest {
  productName: string;
  quantity: number;
  selectedAttributes: Attribute[];
  deliveryMethod: string;

  constructor(
    productName: string,
    quantity: number,
    selectedAttributes: Attribute[],
    deliveryMethod: string
  ) {
    this.productName = productName;
    this.quantity = quantity;
    this.selectedAttributes = selectedAttributes;
    this.deliveryMethod = deliveryMethod;
  }
}

File: price-engine/app/lib/price_calculation_request.py
from typing import List
from dataclasses import dataclass
from .product.attribute import Attribute  # Assuming Attribute is in attribute.py or adjust the import path

@dataclass
class PriceCalculationRequest:
    productName: str
    quantity: float
    selectedAttributes: List[Attribute]
    deliveryMethod: str
